### PR TITLE
Fixes #24472 - Identify report source proxy when using LB

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

**Comment by xprazak2** [comment]
Needs [proxy part](https://github.com/theforeman/smart_proxy_openscap/pull/60) and there are changes for [installer](https://github.com/theforeman/puppet-foreman_proxy/pull/443) as well.

**Comment by xprazak2** [comment]
[test foreman_openscap]

File: app/controllers/api/v2/compliance/arf_reports_controller.rb [hunk 1]
```diff
@@ -84,12 +80,27 @@ def find_resource
...
         end
 
+        def find_resources_before_create
+          @asset = ForemanOpenscap::Helper::get_asset(params[:cname], params[:policy_id])
+          if !params[:openscap_proxy_url] && !params[:openscap_proxy_name]
+            msg = _('Failed to upload Arf Report, OpenSCAP proxy name or url not found in params for when uploading report for %s') % @asset.host.name
+            no_proxy_for_host(msg)
+            return
+          end
+          @smart_proxy = SmartProxy.find_by :name => params[:openscap_proxy_name]
```
Comment (ares) [inline app/controllers/api/v2/compliance/arf_reports_controller.rb]: how do we set current org/loc here? should we use `.unscoped` to always find a proxy, context regardless?

File: app/controllers/api/v2/compliance/arf_reports_controller.rb [hunk 2]
```diff
@@ -84,12 +80,27 @@ def find_resource
...
+          @smart_proxy ||= SmartProxy.find_by :url => params[:openscap_proxy_url]
```
Comment (ares) [inline app/controllers/api/v2/compliance/arf_reports_controller.rb]: the same here

File: app/controllers/api/v2/compliance/arf_reports_controller.rb [hunk 3]
```diff
@@ -84,12 +80,27 @@ def find_resource
           instance_variable_set("@arf_report", resource_scope.find(params[:id]))
         end
 
+        def find_resources_before_create
+          @asset = ForemanOpenscap::Helper::get_asset(params[:cname], params[:policy_id])
+          if !params[:openscap_proxy_url] && !params[:openscap_proxy_name]
```
Comment (ares) [inline app/controllers/api/v2/compliance/arf_reports_controller.rb]: this now became hard failure, meaning older proxies without https://github.com/theforeman/smart_proxy_openscap/pull/60 won't be able to upload arf reports, correct? should we rather fallback to asset's host proxy in such case to remain compatible, just with the limitation for LB?

**Comment by xprazak2** [comment]
I do not think this affect templates seeding, so test failures probably unrelated.

**Comment by ares** [comment]
Thanks, works fine, tested with proxy with and without the patch. Merging!

**Pull request merged**